### PR TITLE
Remove Reflect Usage for Metric Names

### PR DESCRIPTION
**Pull request by malnick**
Fixes issues in DCOS-13945 and:
- [x] Remove abusive reflect usage for deriving datapoints
- [x] Add tagging to container metrics
- [x] Add unit values to container metrics

### CODE REVIEW

File: collectors/mesos/agent/container.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// ResourceStatistics message is very large; it defines many fields that are
+// dependent on a feature being enabled in Mesos, and not all of those features
+// are enabled in DC/OS.
+//
+// Therefore, we redefine the resourceStatistics struct here with only the fields
+// dcos-metrics currently cares about, which should be stable for Mesos API v1.
+//
+// For a complete reference, see:
+// https://github.com/apache/mesos/blob/1.0.1/include/mesos/v1/mesos.proto#L921-L1022
+type resourceStatistics struct {
```
Comment (darkonie) [inline collectors/mesos/agent/container.go]: maybe consider export `resourceStatistics` ? we might want to reuse it.

File: collectors/mesos/agent/container.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// poll queries an agent for container-level metrics, such as
+// CPU, memory, disk, and network usage.
+func (c *Collector) getContainerMetrics() error {
+	u := url.URL{
+		Scheme: c.RequestProtocol,
+		Host:   net.JoinHostPort(c.nodeInfo.IPAddress, strconv.Itoa(c.Port)),
+		Path:   "/containers",
+	}
```
Comment (darkonie) [inline collectors/mesos/agent/container.go]: 👍

File: collectors/mesos/agent/agent.go [hunk 3]
```diff
@@ -282,69 +161,3 @@ func getLabelsByContainerID(containerID string, frameworks []frameworkInfo) map[
...
-
-		pts = append(pts, producers.Datapoint{
-			Name:      parsed.String(),
-			Unit:      "", // TODO(roger): not currently an easy way to get units
-			Value:     uv,
-			Timestamp: t.UTC().Format(time.RFC3339Nano),
-		})
-	}
-	return pts
-}
```
Comment (darkonie) [inline collectors/mesos/agent/agent.go]: yay! 👍 🎉

File: collectors/mesos/agent/metrics.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+			FRAMEWORKID:  container.FrameworkID,
+			EXECUTORID:   container.ExecutorID,
+			EXECUTORNAME: container.ExecutorName,
+		}
+
+		c.log.Debugf("Adding tags for container %s:\n%+v", container.ContainerID, dpTags)
+
+		addDps := []producers.Datapoint{
+			producers.Datapoint{
+				Name:  "cpus.user.time",
```
Comment (philipnrmn) [inline collectors/mesos/agent/metrics.go]: Period-separated datapoint names 👌

File: collectors/mesos/agent/container.go (hunk 1 above)
Comment (malnick) [inline collectors/mesos/agent/container.go]: that's an idea, I hadn't really thought much about using this project as a library. If we did,  I'd want to push it into dcos-go (maybe make that dcos/cluster package I keep wanting) to do it?

**Pull request merged**